sql/moz-fx-data-shared-prod/mozilla_vpn_derived/all_subscriptions_v1/query.sql

```diff
@@ -0,0 +1,208 @@
+    created,
+    trial_end,
+    canceled_at,
+    cancel_at,
+    cancel_at_period_end,
+    ended_at,
+    COALESCE(ended_at, CURRENT_TIMESTAMP) AS end_date,
+  FROM
+    mozdata.stripe.subscriptions
+  WHERE
+    status NOT IN ("incomplete", "incomplete_expired")
+),
+stripe_customers AS (
+  SELECT
+    id AS customer_id,
+    mozfun.map.get_key(metadata, "fxa_uid") AS fxa_uid,
+  FROM
+    mozdata.stripe.customers
+),
+stripe_customer_country AS (
+  SELECT
+    customer AS customer_id,
+    LOWER(
+      -- LAST_VALUE(country ORDER BY created)
+      ARRAY_AGG(payment_method_details.card.country ORDER BY created DESC LIMIT 1)[OFFSET(0)]
+    ) AS country,
+  FROM
+    `moz-fx-data-shared-prod`.stripe_external.charges_v1
+  GROUP BY
+    customer_id
+),
+standardized_country AS (
+  SELECT
+    raw_country AS country,
+    standardized_country AS country_name,
+  FROM
+    mozdata.static.third_party_standardized_country_names
+),
+attribution AS (
+  SELECT
+    id AS user_id,
+    fxa_uid,
+    created_at AS user_registration_date,
+    mozfun.map.get_key(attribution, "referrer") AS referrer,
+    mozfun.map.get_key(attribution, "utm_medium") AS utm_medium,
+    mozfun.map.get_key(attribution, "utm_source") AS utm_source,
+    mozfun.map.get_key(attribution, "utm_content") AS utm_content,
+    mozfun.map.get_key(attribution, "utm_campaign") AS utm_campaign,
+    attribution_category,
+    coarse_attribution_category,
+  FROM
+    mozdata.mozilla_vpn.users
+),
+stripe_vpn_plans AS (
+  SELECT
+    plans.id AS plan_id,
+    plans.amount AS plan_amount,
+    plans.billing_scheme,
+    plans.currency AS plan_currency,
+    plans.interval AS plan_interval,
+    plans.product AS product_id,
+    products.name AS product_name,
+  FROM
+    mozdata.stripe.plans
+  LEFT JOIN
+    mozdata.stripe.products
+  ON
+    plans.product = products.id
+  WHERE
+    products.name = "Mozilla VPN"
+),
+vpn_subscriptions AS (
+  SELECT
+    customer_id,
+    subscription_id,
+    plan_id,
+    status,
+    event_timestamp,
+    customer_start_date,
+    subscription_start_date,
+    created,
+    trial_end,
+    canceled_at,
+    cancel_at,
+    cancel_at_period_end,
+    ended_at,
+    end_date,
+    fxa_uid,
+    country,
+    country_name,
+    user_registration_date,
+    referrer,
+    utm_medium,
+    utm_source,
+    utm_content,
+    utm_campaign,
+    attribution_category,
+    coarse_attribution_category,
+    "FXA" AS provider,
+    plan_amount,
+    billing_scheme,
+    plan_currency,
+    plan_interval,
+    product_id,
+    product_name,
+  FROM
+    stripe_subscriptions
+  JOIN -- exclude subscriptions to non-vpn products
+    stripe_vpn_plans
+  USING
+    (plan_id)
+  LEFT JOIN
+    stripe_customers
+  USING
+    (customer_id)
+  LEFT JOIN
+    stripe_customer_country
+  USING
+    (customer_id)
+  LEFT JOIN
+    standardized_country
+  USING
+    (country)
+  LEFT JOIN
+    attribution
+  USING
+    (fxa_uid)
+  UNION ALL
```

Comment:
These two blocks could be split into their own cte statements e.g. `select * from fxa_subscriptions union all select * from apple_iap_subscriptions`.